src/elfpy/markets.py

```diff
@@ -346,23 +376,25 @@ def check_fees(
                 + state_string
             )
 
-    def tick(self, delta_time):
+    def tick(self, delta_time: float) -> None:
         """Increments the time member variable"""
         self.time += delta_time
 
-    def _open_short(self, trade_details):
+    def _open_short(
+        self, agent_action: MarketAction, token_out: TokenType, time_remaining: float
+    ) -> tuple[MarketDeltas, Wallet]:
         """
         take trade spec & turn it into trade details
         compute wallet update spec with specific details
         will be conditional on the pricing model
         """
         trade_results = self.pricing_model.calc_out_given_in(
-            in_=trade_details.trade_amount,
+            in_=agent_action.trade_amount,
             share_reserves=self.share_reserves,
             bond_reserves=self.bond_reserves,
-            token_out=trade_details.token_out,
+            token_out=token_out,
             fee_percent=self.fee_percent,
-            time_remaining=trade_details.stretched_time_remaining,
+            time_remaining=time_remaining,
```

Comment:
the arg name is time_remaining, but I do think it wants stretched_time_remaining. If I understand correctly, then `calc_out_given_in` is technically agnostic to whether the `time_remaining` is stretched or not. But the current iteration of both ElementV1 & Hyperdrive assume stretched time.

Reply:
yup you're right